src/SourceBuild/content/eng/tools/tasks/Microsoft.DotNet.SourceBuild.Tasks.XPlat/UpdateNuGetConfigPackageSourcesMappings.cs

```diff
@@ -95,6 +97,16 @@ public override bool Execute()
 
             XElement pkgSrcMappingClearElement = pkgSrcMappingElement.Descendants().FirstOrDefault(e => e.Name == "clear");
 
+            // Add package source mappings for local package sources
+            foreach (string packageSource in allSourcesPackages.Keys)
+            {
+                // Skip sources with zero package patterns
+                if (allSourcesPackages[packageSource] != null)
```

Comment:
In the other occurrence of this comment, it used a pattern like `allSourcesPackages?.Count > 0` to account for empty lists. Should that be used here?